QStreak/View Controllers/Record List/RecordListViewModel.swift

```diff
@@ -13,23 +13,52 @@ protocol RecordListViewModelDelegate: AnyObject {
 }
 
 class RecordListViewModel {
-    
     // MARK: - Properties
 
+    var records: [Submission?] {
+        didSet { recordsDidChange?(records) }
+    }
+
+    var recordsDidChange: (([Submission?]) -> Void)?
+    var currentPage = 1
+    var totalPages = 2
+
     weak var delegate: RecordListViewModelDelegate?
 
-    // TODO: - Dummy Values
-    let records = [Record(creationDate: Date(), contactCount: 2, activities: [Activity(name: "Grocery Store", activityID: 1)]),
-                   Record(creationDate: Date(), contactCount: 1, activities: [Activity(name: "Grocery Store", activityID: 1)]),
-                   Record(creationDate: Date(), contactCount: 6, activities: [Activity(name: "Grocery Store", activityID: 1)]),
-                   Record(creationDate: Date(), contactCount: 23, activities: [Activity(name: "Grocery Store", activityID: 1)]),
-                   Record(creationDate: Date(), contactCount: 2423, activities: [Activity(name: "Grocery Store", activityID: 1)])]
-    
+    private var isFetchInProgress = false
+    private let sessionProvider = URLSessionProvider()
+
+    init() {
+        self.records = []
+    }
+
+    func fetchRecords() {
+        guard !isFetchInProgress && totalPages >= currentPage else {
+            return
+        }
```

Comment:
Can just be a single line.
```suggestion
        guard !isFetchInProgress && totalPages >= currentPage else { return }
```